mitodl/odl-video-service
# Permissions with moira lists

#### What are the relevant tickets?
Fixes #52 
Fixes #53 
Fixes #146
Fixes #148 
Fixes #149 
Fixes #151 
Fixes #152 
Fixes #153

#### What's this PR do?
Adds additional permission classes and functions to control who can view or edit collections and videos.

#### How should this be manually tested?
- Log in as the superuser
- Create 3 collections and add 1 or more videos to each one
- Go to the admin console
- For collection 1, add 'odl-engineering' to 'VIEW MOIRA LISTS'
- For collection 2, add 'odl-engineering' to 'ADMIN MOIRA LISTS'
- Create a new user who is not a superuser, and fill in the email address field with your MIT email address
- In a private browser window, log in as the new user. 
- You should see only collection 2 in your list of collections.
- For each link you visit, try the corresponding view in the API and make sure the permissions work roughly the same (what collections/videos you can view or edit don't change).
- Go to collection 2's detail page for collection 2, you should be able to see it.  There's no edit form here right now but on the API page you should be able to make edits.
- Go to collection 2's video detail page, you should be able to see & edit it.
- Try to visit the collection 1 detail page (manually enter the URL), you should get a 403 Forbidden error (or 'Not found' on the API page) 
- Try to view the video detail page for collection 1's video, you should be able to view it.  but if you try to edit it the changes shouldn't be saved. 
- Try to visit the collection detail and video detail pages for collection 3.  You should be denied access to both. 
- Create a new collection. Now you should see the new collection and collection 2 in your list.  Add a video to the new collection.
- Go back to the other window where you're logged in as a superuser. Refresh the collections page.  You should see collections 1-3 and the new collection you created as the other user.  Make sure you can view and edit that other user's collection and its video too.
